keep/api/routes/mapping.py

```diff
@@ -79,3 +79,28 @@ def delete_rule(
     session.commit()
     logger.info("Deleted a mapping rule", extra={"rule_id": rule_id})
     return {"message": "Rule deleted successfully"}
+
+
+@router.put("", description="Update an existing rule")
+def update_rule(rule: MappingRuleDtoUpdate,
+                authenticated_entity: AuthenticatedEntity = Depends(AuthVerifier(["write:rules"])),
+                session: Session = Depends(get_session)) -> MappingRuleDtoOut:
+    logger.info("Updating a mapping rule")
+    existing_rule: MappingRule = session.query(MappingRule).filter(
+        MappingRule.tenant_id == authenticated_entity.tenant_id and MappingRule.id == rule.id).first()
+
+    existing_rule.name = rule.name
```

Comment:
Check if existing_rule is not None, raise proper exception if it is

Reply:
I think if rule.name is somehow None, pydantic will throw a 422 UnprocessableIdentity Error.

Reply:
Oh got it. I misunderstood it with the edited name :)